src/pudl_archiver/zenodo/api_client.py

```diff
@@ -110,18 +84,21 @@ def __init__(
         self.upload_key = upload_key
         self.publish_key = publish_key
 
+        self.depositor = ZenodoDepositor(upload_key, publish_key, session, self.testing)
+
         # Map resource name to resource partitions
         self.resource_parts: dict[str, dict] = {}
 
-        self.deposition: Deposition = None
-        self.new_deposition: Deposition = None
+        self.deposition: Deposition | None = None
```

Comment:
More `mypy` stuff.

Separately, I think in the end I'd like these to not be instance variables at all, if I can get away with it - certainly we don't need `self.deposition` - we only use it to get the draft deposition and then we have no use for it anymore. At least we can make these `_private_ish` variables and know that we only reach into them for tests or something.

Reply:
I think this is a good idea, especially given the comment below about trying to delete files from `self.deposition`